samples/java/hibernate/README.md

```diff
@@ -0,0 +1,82 @@
+  -p my-project -i my-instance \
+  -x
+```
+
+## Creating the Sample Data Model
+Run the following command in this directory. Replace the host, port and database name with the actual host, port and
+database name for your PGAdapter and database setup.
+
+```shell
+psql -h localhost -p 5432 -d my-database -f sample-schema.sql
+```
+
+You can also drop an existing data model using the `drop_data_model.sql` script:
+
+```shell
+psql -h localhost -p 5432 -d my-database -f drop-data-model.sql
+```
+
+## Data Types
+Cloud Spanner supports the following data types in combination with `gorm`.
+
+| PostgreSQL Type                        | Hibernate or Java |
+|----------------------------------------|-------------------|
+| boolean                                | boolean           |
+| bigint / int8                          | long              |
+| varchar                                | String            |
+| text                                   | String            |
+| float8 / double precision              | double            |
+| numeric                                | BigDecimal        |
+| timestamptz / timestamp with time zone | LocalDateTime     |
+| bytea                                  | byte[]            |
+| date                                   | LocalDate         |
+
+## Limitations
+The following limitations are currently known:
+
+| Limitation                           | Workaround                                                                                                                                         |
+|--------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|
+| Migrations                           | Cloud Spanner does not support the full PostgreSQL DDL dialect. Automated migrations using `hibernate` are therefore not supported.                |
+| Generated primary keys               | Cloud Spanner does not support `sequences`. Auto-increment primary key is not supported. Remove auto increment annotation for primary key columns. |
+
+
+### Migrations
+Migrations are not supported as Cloud Spanner does not support the full PostgreSQL DDL dialect. It is recommended to
+create the schema manually. Note that PGAdapter does support `create table if not exists` / `drop table if exists`.
+See [sample-schema.sql](src/main/resources/sample-schema-sql) for the data model for this example.
+
+### Generated Primary Keys
+`Sequences` are not supported. Hence, auto increment primary key is not supported and should be replaced with primary key definitions that
+are manually assigned. See https://cloud.google.com/spanner/docs/schema-design#primary-key-prevent-hotspots
+for more information on choosing a good primary key. This sample uses UUIDs that are generated by the client for primary
+keys.
+
+```java
+public class User {
+	// This uses auto generated UUID.
```

Comment:
nit: indentation

Reply:
I didn't follow. This is already indented.